Api/Controllers/ThreadController.cs

```diff
@@ -0,0 +1,37 @@
+using SharpScape.Api.Models;
+using SharpScape.Api.Data;
+using SharpScape.Shared.Dto;
+using SharpScape.Api.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace SharpScape.Api.Controllers;
+
+[ApiController]
+[Route("api/")]
+public class ThreadController : ControllerBase
+{
+    private readonly AppDbContext _context;
+
+    public ThreadController(AppDbContext context)
+    {
+        _context = context;
+    }
+    // Api to return all threads since we dont have category table it will return threads of all categories
+    [HttpGet,Route("threads")]
+    public async Task<ActionResult<IEnumerable<ThreadModel>>> GetThreads()
+    {
+        return await _context.ThreadModels.ToListAsync();
+    }
+    [HttpGet,Route("threads")]
+    // Api to return info about a thread (title, author, body.. no posts since we dont have post table)
+    // @para ThreadId
+    public ActionResult<ThreadModel> GetThreadInfo([FromBody] ThreadDto request)
+    {
+        var threadInfo = _context.ThreadModels.FirstOrDefault(thread => thread.Id == request.Id);
+        if (threadInfo is null){
+            return BadRequest("Thread not found!!");
+        }
+        return Ok(threadInfo);
```

Comment:
```suggestion
    [HttpGet("{id}")]
    // Api to return info about a thread (title, author, body.. no posts since we dont have post table)
    // @para ThreadId
    public ActionResult<ThreadInfoDto> GetThread(Guid id)
    {
        var thread = _context.Threads.FirstOrDefault(thread => thread.Id == id);
        if (thread is null){
            return BadRequest("Thread not found!!");
        }
        
        var threadInfo = new ThreadInfoDto() {
            Id = thread.Id,
            Title = thread.Title,
            Votes = thread.Votes,
            Replies = thread.Replies,
            Views = thread.Views });
        
        return Ok(threadInfo);
```

Reply:
I dont know why you need to create another variable here. The thread already contains all the need info. I assume that you only want to return some info of a thread instead of all?

Reply:
Everything the frontend displays about a Thread needs to be given via DTO, that's why I added all those fields to the DTO.  We're not going to have the frontend make SQL queries to the DB, that's EntityFrameworkCore's job, and it lives on the backend.

Reply:
No I thought the threadinfo already contains these fields.

Reply:
Originally `threadInfo` was of type `ThreadModel`, which is a type the frontend can't use because that's not in `Shared/`